dataloader.go

```diff
@@ -414,7 +414,9 @@ func (b *batcher) batch(originalContext context.Context) {
 	}
 
 	ctx, finish := b.tracer.TraceBatch(originalContext, keys)
-	defer finish(items)
+	defer func() {
```

Comment:
What difference does this make and what exactly are you trying to solve ?
How is `defer finish(items)` any different than what you have ?

Reply:
With `defer finish(items)`, the function will always receive a slice without any element

Reply:
`defer finish(items)`  -- `items` is evaluated when `finish` is deferred, so it's always a empty slice
`defer func(){ finish(items) }()` -- `items` is evaluated when the deferred func is executed, so the subsequent changes will be applied.

Reply:
Ah , makes sense 👍 . Can you please write a unit test for this ? You can probably mock `tracer` as it's an interface and provide your own definition for finish.

Reply:
I am not good at writing unit tests, I'd be glad to receive your comments